libs/sdk-ui-ext/src/internal/dashboardEmbedding/DashboardLayout/utils/test/sizing_test.ts

```diff
@@ -0,0 +1,73 @@
+// (C) 2019-2020 GoodData Corporation
+import chunk from "lodash/chunk";
+import {
+    getDashboardLayoutContentHeightForRatioAndScreen,
+    getDashboardLayoutMinimumWidgetHeight,
+    unifyDashboardLayoutColumnHeights,
+} from "../sizing";
+import { DashboardLayoutForKDWidgetClass } from "../../interfaces/dashboardLayout";
+import { ALL_SCREENS } from "../../../FluidLayout";
+import { allWidgetClasses, dashboardLayoutMock, dashboardRowMock, dashboardWidgetMock } from "../../mocks";
+import { DASHBOARD_LAYOUT_GRID_COLUMNS_COUNT } from "../../constants";
+
+const widgets = allWidgetClasses.map((widgetClass) => dashboardWidgetMock(widgetClass, widgetClass));
+const rows = chunk(widgets, 12).map((widgetsInRow) =>
+    dashboardRowMock(
+        widgetsInRow.map((widget, index) => [widget, { widthAsGridColumnsCount: index, heightAsRatio: 50 }]),
+    ),
+);
+const dashboardLayout = dashboardLayoutMock(rows);
+
+describe("sizing", () => {
+    describe("getDashboardLayoutMinimumWidgetHeight", () => {
+        it.each(allWidgetClasses)("should get default height for %s", (widgetClass) => {
+            expect(getDashboardLayoutMinimumWidgetHeight(widgetClass)).toMatchSnapshot();
+        });
+
+        it("should get default height for unknown widget", () => {
+            expect(
+                getDashboardLayoutMinimumWidgetHeight("unknown" as DashboardLayoutForKDWidgetClass),
+            ).toMatchSnapshot();
+        });
+    });
+
+    describe("unifyDashboardLayoutColumnHeights", () => {
+        it("should unify dashboard layout column heights for various widgets and sizes", () => {
+            expect(unifyDashboardLayoutColumnHeights(dashboardLayout)).toMatchSnapshot();
+        });
+    });
+
+    describe("getDashboardLayoutContentHeightForRatioAndScreen", () => {
+        const allGridWidths = Array.from(new Array(DASHBOARD_LAYOUT_GRID_COLUMNS_COUNT)).map((_, i) => i);
+        const createTestCasesForRatio = (ratio: number) =>
+            ALL_SCREENS.reduce((acc: [string, number, number][], screen) => {
+                return allGridWidths.reduce((acc2, width) => {
```

Comment:
Nitpick: this whole thing could probably be a flatMap, it would make the code significantly clearer.